client/udp_buffer_pool.go

```diff
@@ -1,13 +1,14 @@
-package shadowsocks
+package client
 
-import "sync"
+import (
+	"sync"
 
-// maxUDPBufferSize is the maximum UDP packet size in bytes.
-const maxUDPBufferSize = 16 * 1024
+	"github.com/Jigsaw-Code/outline-ss-server/shadowsocks"
+)
 
 var pool = sync.Pool{
 	New: func() interface{} {
-		return make([]byte, maxUDPBufferSize)
+		return make([]byte, shadowsocks.MaxUDPPacketSize)
```

Comment:
IIUC, the UDP buffer size is not changing in this PR. However, it may be beneficial to keep as small as possible. As reference, the VPN MTU is 1500 bytes and go-tun2socks uses a default buffer size of 2KB.

Reply:
`maxUDPBufferSize` was 16 KB, but `shadowsocks.MaxUDPPacketSize` is 64 KB, so I think this changes the size of slices we allocate here.

This shouldn't cause a huge effect (there should only be a few of these buffers due to the use of `sync.Pool`) but I didn't want it to change without being noticed.

Reply:
You're right. I would lean toward keeping the 16KB buffer until we profile memory usage in the client.